include/boost/numeric/ublas/tensor/static_extents.hpp

```diff
@@ -114,34 +136,15 @@ template<typename T> struct static_product;
 template<typename T> 
 inline static constexpr auto const static_product_v = static_product<T>::value;
 
-template<typename ExtentsType, ExtentsType E0, ExtentsType... E>
-struct static_product< basic_static_extents<ExtentsType, E0, E...> >{
-  static constexpr auto const value = E0 * static_product_v< basic_static_extents<ExtentsType, E...> >;
-};
-
-template<typename ExtentsType, ExtentsType E0>
-struct static_product< basic_static_extents<ExtentsType, E0> >{
-  static constexpr auto const value = E0 ;
-};
+template<typename ExtentsType, ExtentsType... Es>
+struct static_product< basic_static_extents<ExtentsType, Es...> >
+  : std::integral_constant< ExtentsType, (... * Es) >
```

Comment:
Nice, use of fold expressions.

Reply:
I did not use it earlier because of `msvc` which was giving the error. Now we have changed the 'msvc' version and they have implemented the fold expression I can use it.